src/main/java/com/google/cloud/tools/appengine/cloudsdk/PathResolver.java

```diff
@@ -85,8 +94,33 @@ private static void getLocationsFromPath(List<String> possiblePaths) {
         if (path.endsWith("google-cloud-sdk" + File.separator + "bin")) {
           possiblePaths.add(path.substring(0, path.length() - 4));
         }
+
+        Path possibleLink = Paths.get(path, "gcloud");
+        if (Files.isSymbolicLink(possibleLink)) {
+          getLocationsFromLink(possiblePaths, possibleLink);
+        }
+      }
+    }
+  }
+
+  @VisibleForTesting
+  // resolve symlinks to a path that could be the bin directory of the cloud sdk
+  static void getLocationsFromLink(List<String> possiblePaths, Path link) {
+    try {
+      Path resolvedLink = link.toRealPath();
+      Path possibleBinDir = resolvedLink.getParent();
```

Comment:
It may be possible that `getParent()` doesn't work as you expected: http://stackoverflow.com/a/1099370/1701388. Probably it's better to do `getCanonicalPath()` before it.